DiscordBot/Commands/EventCommands.cs

```diff
@@ -79,15 +86,123 @@ public async Task CreateEvent(CommandContext context, InteractivityModule intera
+            if (discordEmbed == null)
+            {
+                return;
+            }
+
+            await context.RespondAsync($"{context.Member.Mention}", embed: discordEmbed);
+            await context.RespondAsync(
+                $"{context.Member.Mention} - what field do you want to edit? (``name``, ``description``, ``time``)\n" +
+                "``save`` to save changes.");
+            var editField = await GetUserResponse(context, interactivity, EventFields);
+            if (editField == null)
+            {
+                return;
+            }
+
+            await EditEventField(context, interactivity, eventKey.Value, editField, discordEmbed);
+        }
+
+        private static async Task EditEventField(
+            CommandContext context,
+            InteractivityModule interactivity,
+            int eventKey,
+            string editField,
+            DiscordEmbedBuilder discordEmbed)
+        {
+            string? newName = null;
+            string? newDescription = null;
+            DateTimeOffset? newTime = null;
+
+            switch (editField)
+            {
+                case "name":
+                    await context.RespondAsync($"{context.Member.Mention} - enter the new event name.");
+                    newName = await GetUserResponse(context, interactivity);
+                    if (newName == null)
+                    {
+                        return;
+                    }
+
+                    discordEmbed.Title = newName;
+                    break;
+                case "description":
+                    await context.RespondAsync($"{context.Member.Mention} - enter the new description.");
+                    newDescription = await GetUserResponse(context, interactivity);
+                    if (newDescription == null)
+                    {
+                        return;
+                    }
+
+                    discordEmbed.Description = newDescription;
+                    break;
+                case "time":
+                    await context.RespondAsync($"{context.Member.Mention} - enter the new event time.");
+                    newTime = await GetUserTimeResponse(context, interactivity);
+                    if (newTime == null)
+                    {
+                        return;
+                    }
+
+                    discordEmbed.Timestamp = newTime.Value.DateTime;
+                    break;
+            }
+
+            try
+            {
+                var eventsSheetService = context.Dependencies.GetDependency<IEventsSheetsService>();
+                await eventsSheetService.EditEventAsync(eventKey, newDescription, newName, newTime?.DateTime);
+            }
+            catch (EventNotFoundException)
+            {
+                await context.RespondAsync($"{context.Member.Mention} - operation stopped: event not found");
+            }
+
+            await context.RespondAsync($"{context.Member.Mention} - changes saved!", embed: discordEmbed);
```

Comment:
At the moment this would print event though the event wasn't found I believe, which doesn't seem right?